public/jobs.js

```diff
@@ -0,0 +1,355 @@
+async function buildJobsTable(jobsTable, jobsTableHeader, token, message) {
```

Comment:
Hm, I take issue with the organization of this script -- feels like spaghetti code to me, but I've seen it in several students' work at this point so I know it comes from the course. 

So let's discuss! What would you do to improve the structure of this script?